go/vt/vtgate/vindexes/cfc.go

```diff
@@ -0,0 +1,336 @@
+		if err != nil {
+			return nil, err
+		}
+		out[i] = key.DestinationKeyspaceID(v)
+	}
+	return out, nil
+}
+
+// PrefixVindex switches the vindex to prefix mode
+func (vind *CFC) PrefixVindex() SingleColumn {
+	return vind.prefixVindex
+}
+
+type prefixCFC struct {
+	name string
+	cfc  *CFC
+}
+
+func (vind *prefixCFC) String() string {
+	return vind.name
+}
+
+func (vind *prefixCFC) Cost() int {
+	return 1
+}
+
+func (vind *prefixCFC) IsUnique() bool {
+	return false
+}
+
+func (vind *prefixCFC) NeedsVCursor() bool {
+	return false
+}
+
+// Verify returns true if ids maps to ksids.
+func (vind *prefixCFC) Verify(vc VCursor, ids []sqltypes.Value, ksids [][]byte) ([]bool, error) {
+	// prefixCFC is only active for 'LIKE' expr. Verify is used by 'INSERT' so we re-use CFC.Verify()
+	return vind.cfc.Verify(vc, ids, ksids)
+}
+
+// Map can map ids to key.Destination objects.
+func (vind *prefixCFC) Map(cursor VCursor, ids []sqltypes.Value) ([]key.Destination, error) {
+	out := make([]key.Destination, len(ids))
+	for i, id := range ids {
+		value := id.ToBytes()
+		prefix := findPrefix(value)
+		begin, err := vind.cfc.computeKsid(prefix, true)
+		if err != nil {
+			return nil, err
+		}
+		out[i] = key.NewKeyRangeFromPrefix(begin)
+	}
+	return out, nil
+}
+
+// findPrefix returns the 'prefix' of the string literal in LIKE expression.
+// The prefix is the prefix of the string literal up until the first unescaped
+// meta character (% and _). Other escape sequences are escaped according to
+// https://dev.mysql.com/doc/refman/8.0/en/string-literals.html.
+func findPrefix(str []byte) []byte {
+	buf := new(bytes.Buffer)
+L:
+	for len(str) > 0 {
+		n := len(str)
+		p := bytes.IndexAny(str, `%_\`)
+		if p < 0 {
+			buf.Write(str)
+			break
+		}
+		buf.Write(str[:p])
+		switch str[p] {
+		case '%', '_':
+			// prefix found
+			break L
+		// The following is not very efficient in dealing with too many
+		// continuous backslash characters, e.g. '\\\\\\\\\\\\\%', but
+		// hopefully it's the less common case.
+		case '\\':
+			if p == n-1 {
+				// backslash is the very last character of a string, typically
+				// this is an invalid string argument. We write the backslash
+				// anyway because Mysql can deal with it.
+				buf.WriteByte(str[p])
+				break L
+			} else if decoded := sqltypes.SQLDecodeMap[str[p+1]]; decoded != sqltypes.DontEscape {
+				buf.WriteByte(decoded)
+			} else {
+				buf.WriteByte(str[p+1])
+			}
+			str = str[(p + 2):n]
+		}
+	}
+	return buf.Bytes()
+}
+
+// we don't use the full hashed value because it's very long.
+// keyrange resolution is done via comparing []byte so longer
+// keyspace ids have performance impact.
+func md5hash(in []byte) []byte {
+	n := len(in)
+	out := vMD5Hash(in)
+	if n < len(out) {
+		return out[:n]
+	}
+	return out
+
+}
+
+// same here
+func xxhash64(in []byte) []byte {
+	out := vXXHash(in)
+	n := len(in)
+	if n < 8 {
+		return out[:n]
+	}
+	return out
+}
```

Comment:
The goal is to return the same amount of bytes as the incoming value when possible. The key range solution involves `bytes.Compare` which is linear to slice size. If the incoming component has two bytes, we'll return two byte hash. 

The reason to introduce xxhash is because it's faster than md5, https://medium.com/asecuritysite-when-bob-met-alice/when-fast-just-isnt-enough-nobody-inspects-the-spammish-repetition-3458a96aa04e